main.go

```diff
@@ -355,7 +356,13 @@ func setupReconcilers(ctx context.Context, mgr ctrl.Manager) {
 			RuntimeClient:             runtimeClient,
 			UnstructuredCachingClient: unstructuredCachingClient,
 			WatchFilterValue:          watchFilterValue,
-		}).SetupWithManager(ctx, mgr, concurrency(clusterTopologyConcurrency)); err != nil {
+		}).SetupWithManager(ctx, mgr, controller.Options{
+			MaxConcurrentReconciles: clusterTopologyConcurrency,
+			RateLimiter: workqueue.NewMaxOfRateLimiter(
+				workqueue.DefaultControllerRateLimiter(),
```

Comment:
The big issue here is how this new rate limiter interacts with the exponential backoff cache. Right now this set-up returns a requeue which (I'm pretty sure) also triggers exponential backoff. 

At some point if there's a flood of events that would cause the time between reconciles (with no error state) to get very high.

This is definitely an argument to remove the rate limiter here and use RetryAfter behaviour instead. i.e. retryAfter TTL.

Reply:
Is the current idea to implement both? The change in the reconciler + setting the RateLimiter here?

Reply:
IMO this should be something we look into fixing in Controller Runtime itself, wdyt?

Reply:
It would be great to have the feature at that level - I'll open an issue there to track it.